pkg/log/zap/flags.go

```diff
@@ -145,6 +131,59 @@ func (v levelValue) Type() string {
 	return "level"
 }
 
+type stackLevelValue struct {
+	set   bool
+	level zapcore.Level
+}
+
+func (v *stackLevelValue) Set(l string) error {
+	v.set = true
+	lvl, err := intLogLevel(l)
+	if err != nil {
+		return err
+	}
+
+	v.level = zapcore.Level(int8(lvl))
+	return nil
+}
+
+func (v stackLevelValue) String() string {
+	if v.set {
+		return v.level.String()
+	}
+
+	return "error"
```

Comment:
If I don't set the flag `--zap-stacktrace-level` then default value of `stackLevelValue.level` will be the zero value for the type [`zapcore.Level`](https://github.com/uber-go/zap/blob/master/zapcore/level.go#L32:6) which is just 0.
And that translates to level `info`.

So it's only the string output that will list the default as `error`. The actual stacktrace level that gets set as part of the zap options will end up being `info` or 1.
```Go
c.opts = append(c.opts, zap.AddStacktrace(c.stackTraceLevel))
```

Reply:
@hasbro17 I think this may be right. If `--zap-stacktrace-level` is not set and `--zap-devel` is false (the default), then we set `c.stackTraceLevel` to `zap.ErrorLevel` at [pkg/log/zap/logger.go:77](https://github.com/operator-framework/operator-sdk/blob/1cc229face270e69f03b05a672f2eb47349af865/pkg/log/zap/logger.go#L77) and then we skip using `stacktraceLevel.level` in line 82.

Reply:
You're right, I completely missed line 77 🤦‍♂  Should be alright in that case it seems.